secboot/secboot_sb_test.go

```diff
@@ -583,40 +582,37 @@ func (s *secbootSuite) TestUnlockVolumeUsingSealedKeyIfEncrypted(c *C) {
 		devicePath := filepath.Join("/dev/disk/by-partuuid", partuuid)
 
 		keyPath := filepath.Join("test-data", "keyfile")
-		kd, err := sb_tpm2.NewKeyDataFromSealedKeyObjectFile(keyPath)
-		c.Assert(err, IsNil)
-		expectedID, err := kd.UniqueID()
-		c.Assert(err, IsNil)
 
-		restore = secboot.MockSbActivateVolumeWithKeyData(func(volumeName, sourceDevicePath string, keyData *sb.KeyData, options *sb.ActivateVolumeOptions) (sb.SnapModelChecker, error) {
+		restore = secboot.MockSbActivateVolumeWithKeyData(func(volumeName, sourceDevicePath string, authRequestor sb.AuthRequestor, kdf sb.KDF, options *sb.ActivateVolumeOptions, keys ...*sb.KeyData) error {
+
 			c.Assert(volumeName, Equals, "name-"+randomUUID)
 			c.Assert(sourceDevicePath, Equals, devicePath)
-			c.Assert(keyData, NotNil)
-			uID, err := keyData.UniqueID()
-			c.Assert(err, IsNil)
-			c.Check(uID, DeepEquals, expectedID)
+			c.Assert(keys, HasLen, 1)
+			c.Assert(keys[0], NotNil)
```

Comment:
can't we do anything more to check that keys[0] is what we expect?

Reply:
I have mocked the call to MockSbActivateVolumeWithKeyData to check the path and capture what pointer is created and verify it after.